### ISSUE TITLE
[Bug]: DetailList onItemInvoked will be incorrectly triggered on group level 

### ISSUE THREAD

**Issue by jarvan-jiang** [microsoft/fluentui#32039]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 2.61 GB / 15.73 GB
  Browsers:
    Chrome: 126.0.6478.182
  npmPackages:
    @fluentui/date-time-utilities: 8.5.1 => 8.5.1
    @fluentui/font-icons-mdl2: 8.5.25 => 8.5.25
    @fluentui/foundation-legacy: 8.2.10 => 8.2.10
    @fluentui/merge-styles: 8.5.2 => 8.5.2
    @fluentui/react: 8.112.2 => 8.112.2
    @fluentui/react-focus: 8.8.32 => 8.8.32
    @fluentui/react-hooks: 8.6.30 => 8.6.30
    @fluentui/react-portal-compat-context: 9.0.9 => 9.0.9
    @fluentui/react-window-provider: 2.2.15 => 2.2.15
    @fluentui/set-version: 8.2.11 => 8.2.11
    @fluentui/style-utilities: 8.9.18 => 8.9.18
    @fluentui/theme: 2.6.36 => 2.6.36
    @fluentui/utilities: 8.13.19 => 8.13.19
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/jarvan-jiang/pen/QWXyONd

## Actual Behavior
The issue both happened in normal detaillist and grouplistV2.
Double click the empty area of group level, will trigger the onItemInvoked event with the first child of the group.
![image](https://github.com/user-attachments/assets/c0a23574-b994-420a-80c4-59a225519015)

## Expected Behavior
Double click any area of group level, should not trigger onItemInvoked event.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by spmonahan** [comment]
@jarvan-jiang I'm not able to repro this in Edge, Chrome or Firefox on Windows 11. I've attached a screen recording. Am I missing a step?

https://github.com/user-attachments/assets/c0172cfd-f2c7-4d2e-b94b-636ce44cf992

**Comment by jarvan-jiang** [comment]
Sorry  @spmonahan , I accidently updated the code while finding alternative solutions. I have revert the changes and it should be able to reproduce now.

**Comment by spmonahan** [comment]
After digging into this I believe the existing behavior is the intended behavior. When invoking an item the item index is looked up via `_getItemIndex()` which queries the [`data-selection-index` attribute](https://github.com/microsoft/fluentui/blob/master/packages/react/src/utilities/selection/SelectionZone.tsx#L824).

This attribute is set on headers which is why I think it's intended:

<img width="1290" alt="msedge_Pzl0aR4Q2i" src="https://github.com/user-attachments/assets/e46c39b7-dca1-4951-93f1-06d2d62faf17">

Of course this doesn't mean we cannot implement the expected behavior as well. To do this I think we'll need:

1. A way to distinguish headers from items
2. A new prop on `SelectionZone` to control "header invocation" (it should be enabled by default to maintain the current default behavior)

**Comment by jarvan-jiang** [comment]
> After digging into this I believe the existing behavior is the intended behavior. When invoking an item the item index is looked up via `_getItemIndex()` which queries the [`data-selection-index` attribute](https://github.com/microsoft/fluentui/blob/master/packages/react/src/utilities/selection/SelectionZone.tsx#L824).
> 
> This attribute is set on headers which is why I think it's intended:
> 
> <img alt="msedge_Pzl0aR4Q2i" width="1290" src="https://private-user-images.githubusercontent.com/93940821/354686826-e46c39b7-dca1-4951-93f1-06d2d62faf17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MzQzMjEsIm5iZiI6MTcyMjkzNDAyMSwicGF0aCI6Ii85Mzk0MDgyMS8zNTQ2ODY4MjYtZTQ2YzM5YjctZGNhMS00OTUxLTkzZjEtMDZkMmQ2MmZhZjE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDA4NDcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0MzRjYzBmYjljMDhiYjQ1NzYxZGMxMTE3NzJkNGFmZTVjY2ViZTZkYjRhZjMzZjZiNzdhYTdlY2RmNDEyZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IbymYaCOgzVAs7KiV_4DOVThb98azXqcsI1KObAZzgk">
> Of course this doesn't mean we cannot implement the expected behavior as well. To do this I think we'll need:
> 
> 1. A way to distinguish headers from items
> 2. A new prop on `SelectionZone` to control "header invocation" (it should be enabled by default to maintain the current default behavior)

Yeah, that's exactly I found, because we don't have that issue in old version, so I did some research and found the property was added in this [commit](https://github.com/microsoft/fluentui/commit/7c797fb44915964da8f764a93557f2b5c4c7067f).
However, the item passed to `onItemInvoked` is incorrectly set to the row item, so like you said, there is no way to distinguish if the event comes from group or row then caused twice trigger. The reason wrong data passed to the data I believe related to the `selection`  getting wrong data. I did not go further because it's seems related to more scopes.

The temp solution is to wrap a container with  for groupheader, and set ` data-selection-invoke="true"`.

**Comment by spmonahan** [comment]
> The temp solution is to wrap a container with for groupheader, and set  data-selection-invoke="true".

Just confirming, you have a workaround that unblocks you?

**Comment by jarvan-jiang** [comment]
> > The temp solution is to wrap a container with for groupheader, and set  data-selection-invoke="true".
> 
> Just confirming, you have a workaround that unblocks you?

Yes, there is a workaround to wrap the header in container with `data-selection-invoke="true"`.

**Comment by spmonahan** [comment]
Closing this since there is a workaround.
